### PR TITLE
A basic admin frontend and first chance at a review

### PR DESCRIPTION
**Pull request by jonuy**
This is a first look at a potential new direction to take the SMS app. This app would serve the purpose of both 1. providing the additional logic that interfaces with Mobile Commons and 2. offering a frontend to administer the configuration changes needed.

As a proof of concept, I'm trying to implement simple logic from the [current app](https://github.com/DoSomething/ds-mdata-responder) that parses a message from a user and then opts them into another path depending on whether or not their response was a "yes" or a "no."
#### What's this PR do?

This PR specifically addresses the frontend admin part of the problem. See the next section for previous work done that replicates the current app's logic.

It might be easiest to just check what routes got added in **routes.js** and see its corresponding logic in **ConfigController.js**.
- `'/config': 'ConfigController.list'`: This view displays the list of current configurations
- `'get /config/new': 'ConfigController.createView'`: This displays the form for creating a new configuration
- `'post /config/new': 'ConfigController.createSubmit'`: Save the new configuration to the database
- `'get /config/:campaignId': 'ConfigController.editView'`: This displays the form for updating an already existing configuration. The form should be filled in with the existing values when you get to this page.
- `'post /config/:campaignId': 'ConfigController.editSubmit'`: Saves the configuration update to the database
##### Questions:
- Is there a better, more standard way I could have structured this? All the different custom actions in ConfigController feels a little bit weird to me. But then, I'm not exactly sure what I should be doing instead.
- I also have questions about how to properly include Forge into the repo.
#### What else should I review?

This is the previous work done in this repo to replicate the yes/no logic: https://github.com/DoSomething/gambit-sails-prototype/commit/c70e4a2ac1d8a5573cdea735059da3d65cf234f4

Key things to note:
- **api/models/YesNoConfig.js**: This is a replication of [this](https://github.com/DoSomething/ds-mdata-responder/blob/dev/app/lib/ds-routing/config/yesNoPathsConfigModel.js) but could now feasibly be swapped out into a non-MongoDB db should we want to. (Also, ignore the `incomingOptInPathId.primaryKey=true`. I didn't know what I was doing. That's been removed in this PR.)  
- **api/controllers/MobileCommonsController.js**: In the current app, the routes and logic were spread out over different modules. For the foreseeable future, we'll just need to maintain 3 routes for mDatas to hit. The plan is for them all to live in here.  
- **api/services/MobileCommonsService.js**: The 3 main API calls we use are `optin`, `optout` and `profile_update`. Their implementations would be in here and, as a service, be accessible from anywhere in the app.
##### Questions:
- Er, I guess the main thing is does this make sense? I ultimately won't be the one maintaining this so it needs to be easy for someone else to know what they're seeing when they get in here.
#### Next Steps
- Implement the rest of our SMS campaign functionality, both on the mData logic side and the admin configuration side
- Add an authentication gate so only authorized users can use the tool

**cc:** @DFurnes @aaronschachter @sergii-tkachenko 

Thanks for taking a look! Let me know if you've got any thoughts about what you see. I could also schedule a Hangout to review things in person if that makes sense.

### CODE REVIEW

File: api/controllers/ConfigController.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        id: 'new',
+        config: null
+      }
+    );
+  },
+
+  /**
+   * Handle the submission request to create a new campaign configuration.
+   */
+  createSubmit: function(req, res) {
```
Comment (DFurnes) [inline api/controllers/ConfigController.js:41]: I'm not sure if this is also a Sails convention or not, but Laravel (and I think Rails?) use `index`, `create` (the view for creating a new resource), `store` (storing a new resource), `show`, `edit` (the view for editing a new resource), `update` (saving an updated resource), and `destroy` for the resource controller methods for each HTTP verb.

File: assets/styles/gambit.css [hunk 2]
```diff
@@ -0,0 +1,36 @@
+body {
```
Comment (DFurnes) [inline assets/styles/gambit.css]: You can probably use the [Chrome](http://forge.dosomething.org/#chrome) pattern from Forge for this!

File: assets/styles/gambit.css [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  background: white;
+}
+
+main {
```
Comment (DFurnes) [inline assets/styles/gambit.css:1]: And this looks like a [Container](http://forge.dosomething.org/#container)!

File: api/controllers/ConfigController.js (hunk 1 above)
Comment (sergiitk) [inline api/controllers/ConfigController.js:41]: Agree with Dave, here's good Rails doc on method names:
http://guides.rubyonrails.org/routing.html#crud-verbs-and-actions

Comment (DFurnes) [inline api/controllers/ConfigController.js:41]: Hmm, it looks like this [Blueprints](http://sailsjs.org/documentation/concepts/blueprints) feature could be helpful for this! :boat:

**Comment by DFurnes** [comment]
Looks super cool @jonuy! :100:

File: api/controllers/ConfigController.js (hunk 1 above)
Comment (jonuy) [inline api/controllers/ConfigController.js:41]: Yea, I started going down this route at the end of yesterday and just couldn't figure out why `show` wasn't working. I'll keep poking around on it later, but agree that using conventional method names would be ideal.

**Comment by jonuy** [comment]
Merging this in. Will make more changes, like trying to figure out the controller methods, in future PRs.

**Pull request merged**